fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -720,6 +681,31 @@ public Node getNodeVersion(final String label) {
 
     }
 
+    /**
+     * Helps ensure that there are no terminating slashes in the predicate.
+     * A terminating slash means ModeShape has trouble extracting the localName, e.g., for
+     * http://myurl.org/.
+     *
+     * @see <a href="https://jira.duraspace.org/browse/FCREPO-1409"> FCREPO-1409 </a> for details.
+     */
+    private Optional<Quad> checkInvalidPredicates(final UpdateRequest request) {
```

Comment:
Do you mean: why a `Quad` instead of a predicate `String`? The only reason it's returning an `Optional` like this (instead of, e.g., a `boolean`) is so that the `throw` expression can include the problematic part of the statement in the message.